sled-agent/src/updates.rs

```diff
@@ -0,0 +1,117 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Management of per-sled updates
+
+use crate::nexus::NexusClient;
+use omicron_common::api::internal::nexus::{
+    UpdateArtifact, UpdateArtifactKind,
+};
+use std::path::PathBuf;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("I/O Error: {0}")]
+    Io(#[from] std::io::Error),
+
+    #[error("Failed to contact nexus: {0}")]
+    Nexus(anyhow::Error),
+
+    #[error("Failed to read response from Nexus: {0}")]
+    Response(reqwest::Error),
+}
+
+pub async fn download_artifact(
+    artifact: UpdateArtifact,
+    nexus: &NexusClient,
+) -> Result<(), Error> {
+    match artifact.kind {
+        UpdateArtifactKind::Zone => {
+            let directory = PathBuf::from("/var/tmp/zones");
```

Comment:
Yeah, we have this in the demo branch we used yesterday. The main reason I'm keeping this as /var/tmp for now is because this will work in integration tests running on pretty much any box. Giving sled-agent some form of configuration for this value (which we can set to a `TempDir` during testing!) is probably the right way forward here.